src/utils/shuffle.js

```diff
@@ -1 +1,19 @@
-module.exports = array => [...array].sort(() => Math.random() - 0.5)
+const shuffle = array => array.sort(() => Math.random() - 0.5)
+
+module.exports = array => {
+  if (!Array.isArray(array)) {
+    throw new TypeError("'array' is not an array")
+  }
+
+  if (array.length < 2) {
+    return array
+  }
+
+  let shuffled = [...array]
+
+  while (shuffled.every((value, index) => value === array[index])) {
+    shuffled = shuffle(shuffled)
+  }
```

Comment:
It might not matter since it's used as an iterator here, but wouldn't `shuffled.some` suffice here?